### PR TITLE
[finishes #163899407] implementation of the feedback

### PR DESCRIPTION
**Pull request by sportview**
##What does this PR do?##

- Validation of parties to prevent duplicate names for parties

- handle for a party not found status to 404

- Working on the Update name for Parties Endpoint

- Ensure that response message tells a user a party was deleted. Remove extra payload for remaining parties.

- Modifying POST endpoint to return a created party, id and type and name for the office created

How the app should be tested 

clone the Repository and navigate to the myapp folder and run the python file and use Postman

The pivotal story associated with activity #163899407

### CODE REVIEW

File: myapp/api/v1/models/office/office.py [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
         self.officelist=officelist
      
     
-        
-        #create parties
-    def addoffice(self):
+    def add_office(self): 
+        for p in self.officelist:
+            if p["office_name"]==self.office_name:
+                return make_response(jsonify({"status":404,"message":"Office with a duplicate name exists"}))
```
Comment (artorious) [inline myapp/api/v1/models/office/office.py:14]: Linting. line a bit too long

File: myapp/api/v1/views/office/office.py [hunk 2]
```diff
@@ -1,31 +1,28 @@
...
-        office_name=data['office_name']
-        office_type=data['office_type']
-        newoffice=Office(office_name,office_type)  
-        newoffice.addoffice()    
-        return make_response(jsonify(data,{"status":200,"message":"office successfully created"}))
-    else:
-        return make_response(jsonify({"status":404,"message":"office not created"}))
-
+def add_office():
+        if request.get_json():
```
Comment (artorious) [inline myapp/api/v1/views/office/office.py:10]: Hi, are you requesting for user data twice?

File: myapp/api/v1/views/office/office.py [hunk 3]
```diff
@@ -1,31 +1,28 @@
...
-        office_type=data['office_type']
-        newoffice=Office(office_name,office_type)  
-        newoffice.addoffice()    
-        return make_response(jsonify(data,{"status":200,"message":"office successfully created"}))
-    else:
-        return make_response(jsonify({"status":404,"message":"office not created"}))
-
+def add_office():
+        if request.get_json():                
+                data=request.get_json()
```
Comment (artorious) [inline myapp/api/v1/views/office/office.py:11]: See the previous comment.

**Comment by artorious** [comment]
Hi, good job on implementing feedback.

File: myapp/api/v1/models/office/office.py (hunk 1 above)
Comment (sportview) [reply myapp/api/v1/models/office/office.py:14]: okey noted

File: myapp/api/v1/views/office/office.py (hunk 2 above)
Comment (sportview) [reply myapp/api/v1/views/office/office.py:10]: no on which module I check it out

**Comment by ewanyonyi** [comment]
That's good of you. Well done

**Pull request merged**